### DISCUSSION TITLE
config help: Setting prPriority on shared configuration

### DISCUSSION THREAD

**Discussion by cgatian** [renovatebot/renovate#12980] [Request Help]
Installation: Self-hosted Github Enterprise
Renovate: v28.15.0

In my local `renovate.json5` config I'd like to set the priority of a specific group provided by a shared configuration

```
  extends: ['myCompany/renovate-config'],
  packageRules: [
    {
      extends: 'myCompany/renovate-config:group-myLibs',
      prPriority: 1
    },
  ]
```

Where in the `myCompany/renovate-config` repo a `group-myLibs.json` file exists defining the group:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "description": ["Group myLib packages together"],
  "packageRules": [
    {
      "extends": "myCompany/renovate-config:monorepo-myLibs",
      "groupName": "My Lib packages"
    }
  ]
}
```

The error I receive is:
> Missing platform config for local preset.

I tried explicitly specifying `local`, `local>myCompany/renovate-config:monorepo-myLibs`, but that' also fails. My config's `extends` works, just not `extends` scoped to `packageRules`.

**Comment by rarkins** [comment]
You should extend the monorepo preset (myCompany/renovate-config:monorepo-myLibs) and not the group preset.

**Reply by cgatian** [reply]
Looks like it's unable to resove?
```
    {
      extends: 'myCompany/renovate-config:monorepo-myLibs',
      prPriority: 1
    },
```
Error:
```
DEBUG: Preset fetch error (repository=myCompany/myRepo)
       "preset": "myCompany/renovate-config:monorepo-myLibs",
       "err": {
         "message": "Missing platform config for local preset.",
         "stack": "Error: Missing platform config for local preset.\n   
```

**Reply by rarkins** [reply]
Sorry, was copy pasting from mobile. You presumably still need the same e.g. local> prefix which was on it before.

**Comment by cgatian** [comment]
Hmm, still didn't work.
```
DEBUG: Preset fetch error (repository=myCompany/myRepo)
       "preset": "local>myCompany/renovate-config:monorepo-myLibs",
       "err": {
         "message": "Missing platform config for local preset.",
         "stack": "Error: Missing platform config for local preset.\n    
```

**Reply by rarkins** [reply]
You said earlier that you had a preset like this which works:

```
"extends": "myCompany/renovate-config:monorepo-myLibs"
```

I can't explain why that wouldn't work if it works for you in a different repo. I can see you're copy and pasting only select parts of your config, so I don't have the full picture. Did you switch it form repo config (e.g. renovate.json) to admin config (e.g. config.js)?

**Reply by cgatian** [reply]
I have two configuration files. The first is the `./github/renovate.js` which gets provided to the GHA.

```js
const config = {
  endpoint: 'https://git.corp.myCompany.com/api/v3/',
  platform: 'github',
  onboarding: false,
  repositories: ['myCompany/testRepo'],
  gitAuthor: 'svc-renovatebot <svc_renovatebot@myCompany.com>'
};
module.exports = config;
```
```
      - name: Self-hosted Renovate
        uses: renovatebot/github-action@v28.15.0
        with:
          configurationFile: .github/renovate.js
```

The second is the `renovate.json5` which exists at the root of the repo.

```json5
{
  $schema: 'https://docs.renovatebot.com/renovate-schema.json',
  extends: ['myCompany/renovate-config', 'myCompany/renovate-config:commonDependencies'],
  baseBranches: ["main"],
  rangeStrategy: 'pin',
  labels: ['dependencies],
  ignoreDeps: ['bootstrap'],
  dependencyDashboard: true,
  packageRules: [
    {
      extends: 'local>myCompany/renovate-config:monorepo-myLibs',
      prPriority: 1
    },
  ]
}
```

The other presets within `extends` in the json5 file seem to work, just not my `packageRules`.

**Reply by rarkins** [reply]
What's the exact contents of `myCompany/renovate-config:monorepo-myLibs`?

**Reply by cgatian** [reply]
@rarkins Thank you so much for taking the time with this.

`monorepo-myLibs.json`
```json
{
  "sourceUrlPrefixes": ["https://git.corp.myCompany.com/myCompany/myLibs"]
}
```

We have another preset which extends this, creating a group.

`group-myLibs.json`
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "description": ["Group common packages together"],
  "packageRules": [
    {
      "extends": "myCompany/renovate-config:monorepo-myLibs",
      "groupName": "Common packages"
    }
  ]
}
```

I am attempting to use the group or monorepo preset in my local config, and set a `prPriority` on it.

**Reply by rarkins** [reply]
I would have expected the following to work:

```
  "packageRules": [
    {
      "extends": "myCompany/renovate-config:monorepo-myLibs",
      "prPriority": 1
    }
  ]
```

And this is what you apparently tried first, although I have some doubts due to the fact you say it's renovate.json5 yet you were showing JS syntax, not JSON(5):

![image](https://user-images.githubusercontent.com/6311784/145225884-317bd7a1-5b18-4d27-af71-f5fe985fd33d.png)

**Reply by cgatian** [reply]
Tried again just to be certain, with no change. It is strange that the `extends` works, but any `packageRules` fail.
I also tried updating to v29.34.0.

renovate.json5
```
{
  $schema: 'https://docs.renovatebot.com/renovate-schema.json',
  extends: ['myCompany/renovate-config'],
  rangeStrategy: 'pin',
  labels: ['dependencies :package:'],
  ignoreDeps: ['bootstrap'],
  dependencyDashboardApproval: false,
  ignorePaths: ['.github/**'],
  packageRules: [
    {
      extends: "myCompany/renovate-config:monorepo-myLibs",
      prPriority: 1
    },
  ],
}
```

Error:
```
DEBUG: Preset fetch error (repository=myCompany/repo-name)
       "preset": "myCompany/renovate-config:monorepo-myLibs",
       "err": {
         "message": "Missing platform config for local preset.",
```

**Reply by rarkins** [reply]
Unfortunately I am out of ideas for why that's happening. If you can reproduce the same error in a public repo, I can try running it through the debugger
